app/views/casa_cases/_volunteer_assignment.html.erb

```diff
@@ -25,7 +25,7 @@
                 <% end %>
               </td>
               <td>
-                <% if assignment.is_active? %>
+                <% if assignment.is_active? && current_user.role == "casa_admin" || assignment.volunteer.supervisor == current_user %>
```

Comment:
`volunteer.supervisor` should be `volunteer&.supervisor` because not all volunteers will have supervisors

Reply:
@compwron i'm pretty sure that the `&` in this case is only verifying that the `volunteer` is present, so if the `volunteer` doesn't have a `supervisor` `volunteer.supervisor` will be `nil`

Reply:
pretty sure the `&` would be needed if not all `assignments` have `volunteers` (but the `assignment` has a `belongs_to` that validates `volunteer` is present)

Reply:
makes sense :)